ejson/src/ejson-payload-converter.ts

```diff
@@ -0,0 +1,54 @@
+// @@@SNIPSTART typescript-ejson-converter-impl
+import {
+  EncodingType,
+  errorMessage,
+  METADATA_ENCODING_KEY,
+  Payload,
+  PayloadConverterWithEncoding,
+  str,
+  u8,
+} from '@temporalio/common';
+import { PayloadConverterError } from '@temporalio/internal-workflow-common';
+import EJSON from 'ejson';
+
+/**
+ * Converts between values and [EJSON](https://docs.meteor.com/api/ejson.html) Payloads.
+ */
+export class EjsonPayloadConverter implements PayloadConverterWithEncoding {
+  public encodingType = 'ejson/plain' as EncodingType;
+
+  public toPayload(value: unknown): Payload | undefined {
+    if (value === undefined) return undefined;
+    let ejson;
+    try {
+      ejson = EJSON.stringify(value);
+    } catch (e) {
+      throw new UnsupportedEjsonTypeError(
```

Comment:
You should have returned `undefined` here and used the composite converter which throws.

Reply:
I think it makes sense as a descriptive error for the last converter in a composite—it's what we do in Json

Reply:
I thought we dropped that for JSON..
It makes it so you can't reorder the converters.

Reply:
I thought I had convinced you, but don't fully remember. I think having a good error message is more impt than ordering flexibility, but maybe we can put the error in Composite.

Right now the `Composite.toPayload @throws` is lying—it always returns a value or undefined. I don't remember if there was as reason for that, but if not, can change to throw descriptive error.

<img width="1115" alt="image" src="https://user-images.githubusercontent.com/251288/159842304-ecab4bd5-33c3-4061-a919-f92a95b7aa48.png">

Reply:
If it threw, I think we'd just have to catch and use `undefined` in a lot of places—we could use a helper

Reply:
`@throws` isn't lying, sorry thought it was `UnsupportedTypeError`

Reply:
The composite converter should be the one that throws not JSON.
This was part of the review.
We should fix this issue before the next release.

Reply:
Fixed in https://github.com/temporalio/sdk-typescript/pull/558